src/3d/qgs3dsceneexporter.cpp

```diff
@@ -606,15 +606,9 @@ QVector<Qgs3DExportObject *> Qgs3DSceneExporter::processLines( Qt3DCore::QEntity
     if ( renderer->primitiveType() != Qt3DRender::QGeometryRenderer::LineStripAdjacency ) continue;
     Qt3DQGeometry *geom = renderer->geometry();
     Qt3DQAttribute *positionAttribute = findAttribute( geom, Qt3DQAttribute::defaultPositionAttributeName(), Qt3DQAttribute::VertexAttribute );
-    Qt3DQAttribute *indexAttribute = nullptr;
-    for ( Qt3DQAttribute *attribute : geom->attributes() )
-    {
-      if ( attribute->attributeType() == Qt3DQAttribute::IndexAttribute )
-      {
-        indexAttribute = attribute;
-        break;
-      }
-    }
+    auto match = std::find_if( geom->attributes().begin(), geom->attributes().end(), []( Qt3DQAttribute * attribute ) { return attribute->attributeType() == Qt3DQAttribute::IndexAttribute; } );
+    Qt3DQAttribute *indexAttribute = *match;
```

Comment:
shouldn't that be (and similary to other occurences of this PR) ?
```suggestion
    Qt3DQAttribute *indexAttribute = match != geom->attributes().end() ? *match : nullptr;
```

Furthermore https://doc.qt.io/qt-5/qt3drender-qgeometry.html#attributes returns a new QVector. Consequently it is likely wrong to compare geom->attributes().begin() and geom->attributes().end() as they might be iterators over different QVector objects (it might work by chance if for example the iterator stores an index). Cf remark "If you want to iterate over these using an STL iterator, you should always take a copy of the container and iterate over the copy." of https://doc.qt.io/qt-5/containers.html#implicit-sharing-iterator-problem

Overall I'm a bit skeptical about the advantages of trying to comply with this cppcheck warning. It doesn't really need to more efficient / more readable code, and the risks of getting it wrong are non neglectable.

Reply:
@rouault , I've had similar thoughts re readability of the code.

@nyalldawson , should we quiet down the std::find_if for now?